starlette/middleware/base.py

```diff
@@ -175,6 +165,8 @@ async def body_stream() -> typing.AsyncGenerator[bytes, None]:
                         if not message.get("more_body", False):
                             break
 
+                await anyio.sleep(0)
```

Comment:
It looks like this is enough to make the switch so we have time to run the `coro` task after the `break` above.

This shouldn't be the way to solve this issue... Also, it breaks one of the tests.

Reply:
doesn't this only work because in the test case in the PR description, the background task that fails isn't really async so a single task switch (via `sleep(0)`) is enough to trigger the exception, but a background task that does await would still not be triggered by this?

Assuming `await self.app` finishes all background tasks, the `coro` that awaits the app here could signal an event that can be awaited here instead.